onchain/permissionless-arbitration/offchain/entrypoint.lua

```diff
@@ -3,120 +3,92 @@ package.path = package.path .. ";/opt/cartesi/lib/lua/5.4/?.lua"
 local initial_hash = m:state().root_hash
-
-local Blockchain = require "blockchain.node"
 local blockchain = Blockchain:new()
 local contract = blockchain:deploy_contract(initial_hash)
 
-
-local p1
-do
-    local CommitmentBuilder = require "computation.commitment"
-    local builder = CommitmentBuilder:new(machine_path)
-    local client = Client:new(blockchain)
-    p1 = Player:new(contract, client, builder, machine_path)
-end
-
-local p2
-do
-    local FakeCommitmentBuilder = require "computation.fake_commitment"
-
-    -- m:run(m.start_cycle + 1)
-    -- local second_hash = m:state().root_hash
-    -- local builder = FakeCommitmentBuilder:new(initial_hash, second_hash)
-
-    local builder = FakeCommitmentBuilder:new(initial_hash)
-    local client = Client:new(blockchain)
-    p2 = Player:new(contract, client, builder, machine_path)
+-- add more player instances here
+local cmds = {
+    string.format([[sh -c "echo $$ ; exec ./offchain/player/honest_player.lua %d %s %s | tee honest.log"]], 1, contract, machine_path),
+    string.format([[sh -c "echo $$ ; exec ./offchain/player/dishonest_player.lua %d %s %s %s | tee dishonest.log"]], 2, contract, machine_path, initial_hash)
+}
+local pid_reader = {}
+local pid_player = {}
+
+for i, cmd in ipairs(cmds) do
+    local reader = io.popen(cmd)
+    local pid = reader:read()
+    pid_reader[pid] = reader
+    pid_player[pid] = i
 end
 
-local i = 0
+-- gracefully end children processes
+setmetatable(pid_reader, {
+    __gc = function(t)
+        helper.stop_players(t)
+    end
+})
+
+local no_active_players = 0
+local all_idle = 0
+local client = Client:new(1)
+local last_ts = [[01/01/2000 00:00:00]]
 while true do
-    print(string.format("\n\n### ROUND %d ###\n", i))
-
-    print "Player 1 react"
-    if p1:react() then break end
-
-    print ""
-
-    print "Player 2 react"
-    if p2:react() then break end
-
-    i = i + 1
+    local players = 0
+
+    for pid, reader in pairs(pid_reader) do
+        local msg_out = 0
+        players = players + 1
+        last_ts, msg_out = helper.log_to_ts(reader, last_ts)
+
+        -- close the reader and delete the reader entry when there's no more msg in the buffer
+        -- and the process has already ended
+        if msg_out == 0 and helper.is_zombie(pid) then
+            helper.log(pid_player[pid], string.format("player process %s is dead", pid))
+            reader:close()
+            pid_reader[pid] = nil
+            pid_player[pid] = nil
+        end
+    end
+
+    if players > 0 then
+        if helper.all_players_idle(pid_player) then
+            all_idle = all_idle + 1
+            helper.rm_all_players_idle(pid_player)
+        else
+            all_idle = 0
+        end
+
+        -- if all players are idle for 10 consecutive iterations, advance blockchain
+        if all_idle == 5 then
+            print("all players idle, fastforward blockchain for 30 seconds...")
+            client:advance_time(30)
```

Comment:
Can you put this `30` in a variable/constant at the top of the file? Like `local FF_TIME = 30`, this way we can easily spot and change this value. Or maybe to some of the `constants` table?